src/live/embed.rs

```diff
@@ -267,13 +267,15 @@ struct KargEmbedAreas {
     regions: Vec<Region>,
     #[serde(skip_serializing)]
     args: String,
+    arch: Option<String>,
 }
 
 #[derive(Deserialize, Serialize)]
 struct KargEmbedInfo {
     default: String,
     files: Vec<KargEmbedLocation>,
     size: usize,
+    arch: Option<String>,
```

Comment:
I'm wondering if changing these structs will introduce a backwards compat issue.. For example if I use coreos-installer that's released today to embed kargs in an ISO and then later try to read those using new coreos-installer (after this merges and new release is out) will it work because no arch was specified?

Reply:
I've checked modified installer & old ISO, old installer & new ISO - it works on x86. But s390x requires new installer and new image.

Reply:
`iso extract minimal-iso` serializes `KargEmbedInfo`, and I believe would write `"arch": null` to an old ISO.  We should probably mark this `#[serde(skip_serializing_if = "Option::is_none")]`.

But also, I think `arch` is an overly indirect way of encoding the desired suffix; see https://github.com/coreos/coreos-assembler/pull/3527#discussion_r1263118197.

Reply:
Right, it's `null`. Marked with `skip_serializing_if`.